src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/run.py

```diff
@@ -40,9 +42,16 @@ def acr_run(cmd,  # pylint: disable=too-many-locals
     _, resource_group_name = validate_managed_registry(
         cmd, registry_name, resource_group_name, RUN_NOT_SUPPORTED)
 
+    platform_os, platform_arch, platform_variant = get_validate_platform(cmd, os_type, platform)
+
+    EncodedTaskRunRequest, FileTaskRunRequest, PlatformProperties = cmd.get_models(
+        'EncodedTaskRunRequest', 'FileTaskRunRequest', 'PlatformProperties')
```

Comment:
nit: move this block back to its original place, so that if error out early, not need to make these calls.